File: Tree/Inorder.java
class Node{
    int data;
    Node left, right;

    Node(int data){
        this.data = data;
        left = right = null;
    }
}

public class Inorder {
    static void inorder_traversal(Node root){
        if(root == null)
            return;

        inorder_traversal(root.left);
        System.out.print(root.data + " ");
        inorder_traversal(root.right);
    }

    public static void main(String[] args) {
        Node root = new Node(1);
        root.left = new Node(2);
        root.right = new Node(3);

        inorder_traversal(root);

    }
    
}

File: Tree/Inorder.py
class Node:
    def __init__(self, data, left=None, right=None):
        self.data = data
        self.left = left
        self.right = right

    def inorder(self, root):
        if root is None:
            return
        self.inorder(root.left)
        print(root.data, end = ' ')
        self.inorder(root.right)


    def main(self):
        root = Node(1)
        root.left = Node(2)
        root.right = Node(3)

        self.inorder(root)

if __name__ == "__main__":
    obj = Node(0)
    obj.main()